### PR TITLE
feature: Save and load settings on the server

### PR DESCRIPTION
**Pull request by cmotelet**
Save and load settings on the server #127
image
image

### CODE REVIEW

**Comment by tkurki** [comment]
I think the new menu texts work pretty well!

There needs to be a separate prompt about enabling security on the server in case it is not there. Now IP just tells you to log in, even if security is not enabled at all.

File: lib/ui/navbar.js [hunk 1]
```diff
@@ -18,8 +21,14 @@ import {
...
   retrieveStartConnected,
   storeColorScheme,
-  retrieveColorScheme
+  retrieveColorScheme,
+  retrieveToken,
+  isPreferredUnits,
+  storePreferredUnits
 } from '../util/localstorage';
+import {
+  retrievePreferredUnits as ssRetrievePreferredUnits
```
Comment (tkurki) [inline lib/ui/navbar.js:30]: why the rename here?

File: lib/ui/navbar.js [hunk 2]
```diff
@@ -363,19 +372,46 @@ export default class IpNavBar extends React.Component {
         "connecting": true,
       });
       SignalkClient.on('connect', () => {
-        SignalkClient.subscribe({context:'vessels.self',subscribe:[{ path:'*'}]});
-        SignalkClient.on('delta', this.props.instrumentPanel.dispatch.bind(this.props.instrumentPanel));
+        // For debug only with npm start
```
Comment (tkurki) [inline lib/ui/navbar.js:375]: Why the token authentication here?

File: lib/ui/navbar.js [hunk 3]
```diff
@@ -363,19 +372,46 @@ export default class IpNavBar extends React.Component {
...
+        if (token) {
+          debug("[connectOrDisconnect] connect with token authentication");
+          SignalkClient.setAuthenticated(token);
+        }
+        // End for debug only
+        debug("[connectOrDisconnect] Waiting for connectionInfo");
+        SignalkClient.on('connectionInfo', (data => {
+          debug("[connectOrDisconnect] connectionInfo received");
+          if (!isPreferredUnits() && this.props.instrumentPanel.getSsAvailable()) {
+            debug("[connectOrDisconnect] storage server is available & no local preferred units");
```
Comment (tkurki) [inline lib/ui/navbar.js:386]: applicationData is the name of the feature on the server, so use it here also?

File: lib/ui/navbar.js (hunk 2 above)
Comment (cmotelet) [reply lib/ui/navbar.js]: I need it for easy debug with `npm start` when IP running at 3001 listening port

File: lib/ui/navbar.js (hunk 3 above)
Comment (cmotelet) [reply lib/ui/navbar.js]: Changed to `applicationData`

File: lib/ui/navbar.js (hunk 1 above)
Comment (cmotelet) [reply lib/ui/navbar.js:30]: I've used prefix `ss` (ServerStorage) as opposed to `ls` (LocalStorage).
The functions are almost the same in localstorage & serverstorage, I prefix them when I have to use both in the same module.
I can replace by `ApplicationData` to speak the same language as the server.
As you want.

**Comment by cmotelet** [comment]
> I think the new menu texts work pretty well!
> 
> There needs to be a separate prompt about enabling security on the server in case it is not there. Now IP just tells you to log in, even if security is not enabled at all.

Updated with `/admin/#/login` for log in and `/admin/#/security/settings` when security not enabled

**Pull request merged**